source/module_io/test/numerical_basis_test.cpp

```diff
@@ -0,0 +1,158 @@
+#include "gtest/gtest.h"
+#include "module_io/numerical_basis_jyjy.h"
+#include "module_base/math_sphbes.h"
+
+using namespace NumericalBasis;
+using ModuleBase::Sphbes;
+
+/***********************************************************
+ *      Unit test of class "TwoCenterIntegrator"
+ ***********************************************************/
+/**
+ * Tested functions:
+ *
+ * - indexgen
+ *   generates a vector of composite index (itype, iatom, l, m)
+ *   from given number of atoms and maximum angular momentum
+ *
+ * - cal_overlap_Sq (static function)
+ *   computes <jy|jy> and <jy|-\nabla^2|jy> with two-center integration
+ *
+ */
+class NumericalBasisTest : public ::testing::Test
+{
+protected:
+    void SetUp();
+    void TearDown();
+
+};
+
+void NumericalBasisTest::SetUp()
+{
+}
+
+void NumericalBasisTest::TearDown()
+{
+}
+
+TEST_F(NumericalBasisTest, indexgen)
+{
+    std::vector<int> natom = {1, 2, 3};
+    std::vector<int> lmax = {2, 1, 2};
+
+    auto index = indexgen(natom, lmax);
+    size_t idx = 0;
+
+    for (size_t it = 0; it < natom.size(); ++it)
+    {
+        for (int ia = 0; ia != natom[it]; ++ia)
+        {
+            for (int l = 0; l <= lmax[it]; ++l)
+            {
+                for (int M = 0; M < 2*l+1; ++M)
+                {
+                    int m = (M % 2 == 0) ? -M/2 : (M+1)/2;
+                    EXPECT_EQ(index[idx], std::make_tuple(int(it), ia, l, m));
+                    ++idx;
+                }
+            }
+        }
+    }
+}
+
+TEST_F(NumericalBasisTest, cal_overlap_Sq)
+{
+    int lmax = 3;
```

Comment:
const int lmax?

Reply:
A const qualifiers has been added in the latest commit.